nemo/collections/asr/parts/submodules/rnnt_loop_labels_computer.py

```diff
@@ -207,20 +230,35 @@ def __init__(
         self._init_confidence_method(confidence_method_cfg=confidence_method_cfg)
         assert self._SOS == self._blank_index  # "blank as pad" algorithm only
 
-        self.use_cuda_graphs = allow_cuda_graphs
-
-        if self.use_cuda_graphs and self.max_symbols is None:
-            logging.warning("Max symbols is None, which is not allowed with Cuda graphs.")
-            self.use_cuda_graphs = False
+        if not allow_cuda_graphs:
+            self.cuda_graphs_mode = None
+        else:
+            # cuda graphs are allowed
+            # check basic requirements for cuda graphs
+            if self.max_symbols is None:
+                logging.warning("Max symbols is None, which is not allowed with Cuda graphs.")
+                self.cuda_graphs_mode = None
+            else:
+                # basic requirements met, need to check while loops
+                try:
+                    check_cuda_python_cuda_graphs_conditional_nodes_supported()
+                    self.cuda_graphs_mode = self.CudaGraphsMode.FULL_GRAPH
+                except (ImportError, ModuleNotFoundError) as e:
+                    logging.warning(
+                        "No conditional node support for Cuda.\n"
+                        "Cuda graphs with while loops are disabled, decoding speed will be slower\n"
+                        f"Reason: {e.msg}"
+                    )
+                    self.cuda_graphs_mode = self.CudaGraphsMode.NO_WHILE_LOOPS
 
-        if self.use_cuda_graphs:
-            try:
-                check_cuda_python_cuda_graphs_conditional_nodes_supported()
-            except ImportError as e:
-                logging.warning(f"No conditional node support. Cuda graphs will be disabled,\n{e.msg}")
-                self.use_cuda_graphs = False
+        self.state = None
+        self.full_graph = None
+        self.separate_graphs = None
 
-        self.state: Optional[LoopLabelsState] = None
+    def force_cuda_graphs_mode(self, mode: Optional[Union[str, CudaGraphsMode]]):
```

Comment:
Do we need this? I don't see any usages of it.

Reply:
Thanks! I forgot to add an explicit test. Fixed.
This is useful for debugging to set `no_graphs` mode (since it's impossible to debug CUDA graphs directly).